Document how to set up B2B module and Payment on account (#259)
front-commerce/developers.front-commerce.com: source/docs/advanced/payments/payment-on-account.md

---
id: payment-on-account
title: Payment on account
---

> Payment on account is a payment method for B2B context that allows companies to make purchases up to the credit limit that is specified in their profile.

Reviewer: ```suggestion
Payment on account is a payment method for B2B contexts. It allows companies to place orders using the credit limit specified in their profile.
```

<blockquote class="feature--new">
_Since version 2.11.0_
</blockquote>

> ## Enable Payment on account

Reviewer: For consistency with other payment methods, I think we should add a new title level with the Platform.

```
## Magento2 B2B

<blockquote class="feature--new">
_Since version 2.11.0_
</blockquote>

### Enable Payment on account
```

<blockquote class="note">
This feature is directly provided by Adobe Commerce. Please refer to [the corresponding documentation page to properly configure it on Magento side](https://docs.magento.com/user-guide/payment/payment-on-account.html).
</blockquote>

### Enable the B2B module

First, you need to [enable and integrate the Front-Commerce B2B module for Magento2](/docs/magento2/b2b.html#Enable-B2B-support).

### Register the Payment on account payment component

1. Override the file that lets you register additional payments components in Front-Commerce
  ```bash
mkdir -p my-module/web/theme/modules/Checkout/Payment/AdditionalPaymentInformation/
cp -u node_modules/front-commerce/src/web/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js my-module/web/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js
  ```
  1. Register the component `PaymentOnAccount` to used for `companycredit` payments in `getAdditionalDataComponent.js`
  ```diff

> +import PaymentOnAccount from "./PaymentOnAccount";

Reviewer 2: where is `./PaymentOnAccount` it is not clear.
maybe add absolute path like `theme/...`

Author: agreed and fixed (it's modeled after the other payment method documentation pages)
